drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -542,13 +746,36 @@ local function do_configure(driver, device)
   end
 end
 
+local function get_endpoints_for_dt(device, device_type)
+  local endpoints = {}
+  for _, ep in ipairs(device.endpoints) do
+    for _, dt in ipairs(ep.device_types) do
+      if dt.device_type_id == device_type then
+        table.insert(endpoints, ep.endpoint_id)
+        break
+      end
+    end
+  end
+  table.sort(endpoints)
+  return endpoints
+end
+
 local function device_added(driver, device)
   local req = im.InteractionRequest(im.InteractionRequest.RequestType.READ, {})
   req:merge(clusters.Thermostat.attributes.ControlSequenceOfOperation:read(device))
   req:merge(clusters.FanControl.attributes.FanModeSequence:read(device))
   req:merge(clusters.FanControl.attributes.WindSupport:read(device))
   req:merge(clusters.FanControl.attributes.RockSupport:read(device))
   device:send(req)
+  local heat_pump_eps = get_endpoints_for_dt(device, HEAT_PUMP_DEVICE_TYPE_ID)
+  local thermostat_eps = get_endpoints_for_dt(device, THERMOSTAT_DEVICE_TYPE_ID)
+  if #heat_pump_eps > 0 then
+    local component_to_endpoint_map = {
+      ["thermostatOne"] = thermostat_eps[1],
+      ["thermostatTwo"] = thermostat_eps[2],
+    }
+    device:set_field(COMPONENT_TO_ENDPOINT_MAP, component_to_endpoint_map, {persist = true})
+  end
```

Comment:
```suggestion
  if #heat_pump_eps > 0 then
    local thermostat_eps = get_endpoints_for_dt(device, THERMOSTAT_DEVICE_TYPE_ID)
    local component_to_endpoint_map = {
      ["thermostatOne"] = thermostat_eps[1],
      ["thermostatTwo"] = thermostat_eps[2],
    }
    device:set_field(COMPONENT_TO_ENDPOINT_MAP, component_to_endpoint_map, {persist = true})
  end
```
Though this is pretty un-extendable as-is. I do think we should extend this logic to permit 1, 2, or more thermostats, even if we only have a profile to support the 2 case for now.